src/main/java/org/openrewrite/java/migrate/lombok/LombokValueToRecord.java

```diff
@@ -0,0 +1,308 @@
+            if (!isClassExpression(expression)) {
+                return false;
+            }
+
+            JavaType.Class classType = (JavaType.Class) expression.getType();
+            if (classType == null) {
+                return false;
+            }
+
+            String methodName = methodInvocation.getName().getSimpleName();
+            String classFqn = classType.getFullyQualifiedName();
+
+            return recordTypeToMembers.containsKey(classFqn)
+                   && methodName.startsWith(STANDARD_GETTER_PREFIX)
+                   && recordTypeToMembers.get(classFqn).contains(getterMethodNameToFluentMethodName(methodName));
+        }
+
+        private static boolean isClassExpression(@Nullable Expression expression) {
+            return expression != null && (expression.getType() instanceof JavaType.Class);
+        }
+
+        private static String getterMethodNameToFluentMethodName(String methodName) {
+            StringBuilder fluentMethodName = new StringBuilder(
+                    methodName.replace(STANDARD_GETTER_PREFIX, ""));
+
+            if (fluentMethodName.length() == 0) {
+                return "";
+            }
+
+            char firstMemberChar = fluentMethodName.charAt(0);
+            fluentMethodName.setCharAt(0, Character.toLowerCase(firstMemberChar));
+
+            return fluentMethodName.toString();
+        }
+
+        private static List<Statement> mapToConstructorArguments(List<J.VariableDeclarations> memberVariables) {
+            return memberVariables
+                    .stream()
+                    .map(it -> it
+                            .withModifiers(Collections.emptyList())
+                            .withVariables(it.getVariables())
+                    )
+                    .map(Statement.class::cast)
+                    .collect(toList());
+        }
+
+        private J.ClassDeclaration addExactToStringMethod(J.ClassDeclaration classDeclaration,
+                                                          List<J.VariableDeclarations> memberVariables) {
+            return classDeclaration.withBody(TO_STRING_TEMPLATE
+                    .apply(new Cursor(getCursor(), classDeclaration.getBody()),
+                            classDeclaration.getBody().getCoordinates().lastStatement(),
+                            classDeclaration.getSimpleName(),
+                            memberVariablesToString(getMemberVariableNames(memberVariables))));
+        }
+
+        private static String memberVariablesToString(Set<String> memberVariables) {
+            return memberVariables
+                    .stream()
+                    .map(member -> String.format(TO_STRING_MEMBER_LINE_PATTERN, member, member))
+                    .collect(Collectors.joining(TO_STRING_MEMBER_DELIMITER));
+        }
+
+        private static JavaType.Class buildRecordType(J.ClassDeclaration classDeclaration) {
+            requireNonNull(classDeclaration.getType(), "Class type must not be null");
```

Comment:
Rather than throwing an exception, it would be better if we just made sure, that the recipe doesn't do anything. Typically, a class should indeed by correctly type attributed. But when this recipe is run together with others, we can't always have too high expectations of what the other recipes do, so it could happen that another recipe gets that wrong. So that this gets detected during development, we could add an `assert` statement instead.

Reply:
addressed